### PR TITLE
remove inline and __forceinline__ from ALPAKA_FN_XXX

### PR DESCRIPTION
**Pull request by BenjaminW3**
Solves a part of #183.

### CODE REVIEW

**Comment by psychocoderHPC** [comment]
I will ckeck if this has negative effects on the performance

**Comment by psychocoderHPC** [comment]
If you remove inline we need a ALPAKA_ACC_INLINE that we can force inline functions on the accelerator side.

**Comment by mxmlnkn** [comment]
Why not just let the user add `inline` where he wants? Are there different kind of inline keywords for different accelerators?

**Comment by BenjaminW3** [comment]
> I will ckeck if this has negative effects on the performance

Your help is greatly appreciated!

**Comment by BenjaminW3** [comment]
CI fails because AppVeyor build times exceed the maximum compilation time (fixed in #193)

File: include/alpaka/core/Common.hpp [hunk 1]
```diff
@@ -75,6 +75,15 @@
     #define ALPAKA_NO_HOST_ACC_WARNING
 #endif
 
+//-----------------------------------------------------------------------------
+//! Macro defining the inline function attribute.
+//-----------------------------------------------------------------------------
+#if defined(ALPAKA_ACC_GPU_CUDA_ENABLED) && defined(__CUDACC__)
+    #define ALPAKA_INLINE __forceinline__
```
Comment (psychocoderHPC) [inline include/alpaka/core/Common.hpp]: Is there a reason why this is called `ALPAKA_INLINE` and not `ALPAKA_FN_INLINE`

**Comment by psychocoderHPC** [comment]
My first check with a complex example [PIConGPU-alpaka](https://github.com/psychocoderHPC/picongpu-alpaka/tree/topic-cupla) shows only small differences in the ptx code with and without `__forceinline__` on CUDA 7.0.

effects:
- in some kernels we need 38 bytes instead of 30 bytes more cmem (IMO no dramatic)
- in some kernel we save one register (IMO great)

with `__forceinline__`
```
[0mptxas info    : 90652 bytes gmem, 364 bytes cmem[3][0m
[0mptxas info    : Compiling entry function '_ZN6alpaka4exec4cuda6detail10cudaKernelISt17integral_constantImLm3EEjN5cupla11CuplaKernelIN5PMacc20kernelCountParticlesINS8_4math2CT6VectorIN4mpl_10integral_cIiLi8EEESF_NSE_IiLi4EEEEEEEEEJNS8_12ParticlesBoxINS8_5FrameINS8_15ParticlesBufferINS8_19ParticleDescriptionIN5boost3mpl6stringILi105ELi0ELi0ELi0ELi0ELi0ELi0ELi0EEESH_NSP_6v_itemIN8picongpu24placeholder_definition219weightingENSS_INST_24placeholder_definition198momentumENSS_INST_24placeholder_definition168positionINST_24placeholder_definition1812position_picENS8_24placeholder_definition1513pmacc_isAliasEEENSP_7vector0INSD_2naEEELi0EEELi0EEELi0EEENSP_6vectorINST_24placeholder_definition2614particlePusherINST_9particles6pusher5BorisES13_EENST_24placeholder_definition255shapeINS1E_6shapes3TSCES13_EENST_24placeholder_definition2913interpolationINST_28FieldToParticleInterpolationIS1L_NST_30AssignedTrilinearInterpolationEEES13_EENST_24placeholder_definition307currentINST_13currentSolver9EsirkepovIS1L_Lj3EEES13_EENST_24placeholder_definition329massRatioINST_24placeholder_definition6513MassRatioIonsES13_EENST_24placeholder_definition3311chargeRatioINST_24placeholder_definition6615ChargeRatioIonsES13_EES16_S16_S16_S16_S16_S16_S16_S16_S16_S16_S16_S16_S16_S16_EENS8_17HandleGuardRegionINS8_9particles8policies17ExchangeParticlesENS2C_15DeleteParticlesEEES17_S17_EESH_Lj3EE29OperatorCreatePairStaticArrayILj256EEENSN_ISR_SH_NSS_INS8_24placeholder_definition139multiMaskENSS_INS8_24placeholder_definition1212localCellIdxES1A_Li0EEELi0EEES29_S2F_S17_NSS_INS8_12NextFramePtrINSD_3argILi1EEEEENSS_INS8_16PreviousFramePtrIS2S_EES17_Li0EEELi0EEEEEEELj3EEEPyNS8_13DefaultFilterINS8_16PositionFilter3DINS8_10TrueFilterEEEEENS8_11AreaMappingILj3ENS8_18MappingDescriptionILj3ESH_EEEEEEEvNS_3VecIT_T0_EET1_DpT2_' for 'sm_37'[0m
[0mptxas info    : Function properties for _ZN6alpaka4exec4cuda6detail10cudaKernelISt17integral_constantImLm3EEjN5cupla11CuplaKernelIN5PMacc20kernelCountParticlesINS8_4math2CT6VectorIN4mpl_10integral_cIiLi8EEESF_NSE_IiLi4EEEEEEEEEJNS8_12ParticlesBoxINS8_5FrameINS8_15ParticlesBufferINS8_19ParticleDescriptionIN5boost3mpl6stringILi105ELi0ELi0ELi0ELi0ELi0ELi0ELi0EEESH_NSP_6v_itemIN8picongpu24placeholder_definition219weightingENSS_INST_24placeholder_definition198momentumENSS_INST_24placeholder_definition168positionINST_24placeholder_definition1812position_picENS8_24placeholder_definition1513pmacc_isAliasEEENSP_7vector0INSD_2naEEELi0EEELi0EEELi0EEENSP_6vectorINST_24placeholder_definition2614particlePusherINST_9particles6pusher5BorisES13_EENST_24placeholder_definition255shapeINS1E_6shapes3TSCES13_EENST_24placeholder_definition2913interpolationINST_28FieldToParticleInterpolationIS1L_NST_30AssignedTrilinearInterpolationEEES13_EENST_24placeholder_definition307currentINST_13currentSolver9EsirkepovIS1L_Lj3EEES13_EENST_24placeholder_definition329massRatioINST_24placeholder_definition6513MassRatioIonsES13_EENST_24placeholder_definition3311chargeRatioINST_24placeholder_definition6615ChargeRatioIonsES13_EES16_S16_S16_S16_S16_S16_S16_S16_S16_S16_S16_S16_S16_S16_EENS8_17HandleGuardRegionINS8_9particles8policies17ExchangeParticlesENS2C_15DeleteParticlesEEES17_S17_EESH_Lj3EE29OperatorCreatePairStaticArrayILj256EEENSN_ISR_SH_NSS_INS8_24placeholder_definition139multiMaskENSS_INS8_24placeholder_definition1212localCellIdxES1A_Li0EEELi0EEES29_S2F_S17_NSS_INS8_12NextFramePtrINSD_3argILi1EEEEENSS_INS8_16PreviousFramePtrIS2S_EES17_Li0EEELi0EEEEEEELj3EEEPyNS8_13DefaultFilterINS8_16PositionFilter3DINS8_10TrueFilterEEEEENS8_11AreaMappingILj3ENS8_18MappingDescriptionILj3ESH_EEEEEEEvNS_3VecIT_T0_EET1_DpT2_[0m
[0m    12288 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads[0m
```
vs. without inline
```
[0mptxas info    : 90733 bytes gmem, 512 bytes cmem[3][0m
[0mptxas info    : Compiling entry function '_ZN6alpaka4exec4cuda6detail10cudaKernelISt17integral_constantImLm3EEjN5cupla11CuplaKernelIN5PMacc20kernelCountParticlesINS8_4math2CT6VectorIN4mpl_10integral_cIiLi8EEESF_NSE_IiLi4EEEEEEEEEJNS8_12ParticlesBoxINS8_5FrameINS8_15ParticlesBufferINS8_19ParticleDescriptionIN5boost3mpl6stringILi105ELi0ELi0ELi0ELi0ELi0ELi0ELi0EEESH_NSP_6v_itemIN8picongpu24placeholder_definition219weightingENSS_INST_24placeholder_definition198momentumENSS_INST_24placeholder_definition168positionINST_24placeholder_definition1812position_picENS8_24placeholder_definition1513pmacc_isAliasEEENSP_7vector0INSD_2naEEELi0EEELi0EEELi0EEENSP_6vectorINST_24placeholder_definition2614particlePusherINST_9particles6pusher5BorisES13_EENST_24placeholder_definition255shapeINS1E_6shapes3TSCES13_EENST_24placeholder_definition2913interpolationINST_28FieldToParticleInterpolationIS1L_NST_30AssignedTrilinearInterpolationEEES13_EENST_24placeholder_definition307currentINST_13currentSolver9EsirkepovIS1L_Lj3EEES13_EENST_24placeholder_definition329massRatioINST_24placeholder_definition6513MassRatioIonsES13_EENST_24placeholder_definition3311chargeRatioINST_24placeholder_definition6615ChargeRatioIonsES13_EES16_S16_S16_S16_S16_S16_S16_S16_S16_S16_S16_S16_S16_S16_EENS8_17HandleGuardRegionINS8_9particles8policies17ExchangeParticlesENS2C_15DeleteParticlesEEES17_S17_EESH_Lj3EE29OperatorCreatePairStaticArrayILj256EEENSN_ISR_SH_NSS_INS8_24placeholder_definition139multiMaskENSS_INS8_24placeholder_definition1212localCellIdxES1A_Li0EEELi0EEES29_S2F_S17_NSS_INS8_12NextFramePtrINSD_3argILi1EEEEENSS_INS8_16PreviousFramePtrIS2S_EES17_Li0EEELi0EEEEEEELj3EEEPyNS8_13DefaultFilterINS8_16PositionFilter3DINS8_10TrueFilterEEEEENS8_11AreaMappingILj3ENS8_18MappingDescriptionILj3ESH_EEEEEEEvNS_3VecIT_T0_EET1_DpT2_' for 'sm_37'[0m
[0mptxas info    : Function properties for _ZN6alpaka4exec4cuda6detail10cudaKernelISt17integral_constantImLm3EEjN5cupla11CuplaKernelIN5PMacc20kernelCountParticlesINS8_4math2CT6VectorIN4mpl_10integral_cIiLi8EEESF_NSE_IiLi4EEEEEEEEEJNS8_12ParticlesBoxINS8_5FrameINS8_15ParticlesBufferINS8_19ParticleDescriptionIN5boost3mpl6stringILi105ELi0ELi0ELi0ELi0ELi0ELi0ELi0EEESH_NSP_6v_itemIN8picongpu24placeholder_definition219weightingENSS_INST_24placeholder_definition198momentumENSS_INST_24placeholder_definition168positionINST_24placeholder_definition1812position_picENS8_24placeholder_definition1513pmacc_isAliasEEENSP_7vector0INSD_2naEEELi0EEELi0EEELi0EEENSP_6vectorINST_24placeholder_definition2614particlePusherINST_9particles6pusher5BorisES13_EENST_24placeholder_definition255shapeINS1E_6shapes3TSCES13_EENST_24placeholder_definition2913interpolationINST_28FieldToParticleInterpolationIS1L_NST_30AssignedTrilinearInterpolationEEES13_EENST_24placeholder_definition307currentINST_13currentSolver9EsirkepovIS1L_Lj3EEES13_EENST_24placeholder_definition329massRatioINST_24placeholder_definition6513MassRatioIonsES13_EENST_24placeholder_definition3311chargeRatioINST_24placeholder_definition6615ChargeRatioIonsES13_EES16_S16_S16_S16_S16_S16_S16_S16_S16_S16_S16_S16_S16_S16_EENS8_17HandleGuardRegionINS8_9particles8policies17ExchangeParticlesENS2C_15DeleteParticlesEEES17_S17_EESH_Lj3EE29OperatorCreatePairStaticArrayILj256EEENSN_ISR_SH_NSS_INS8_24placeholder_definition139multiMaskENSS_INS8_24placeholder_definition1212localCellIdxES1A_Li0EEELi0EEES29_S2F_S17_NSS_INS8_12NextFramePtrINSD_3argILi1EEEEENSS_INS8_16PreviousFramePtrIS2S_EES17_Li0EEELi0EEEEEEELj3EEEPyNS8_13DefaultFilterINS8_16PositionFilter3DINS8_10TrueFilterEEEEENS8_11AreaMappingILj3ENS8_18MappingDescriptionILj3ESH_EEEEEEEvNS_3VecIT_T0_EET1_DpT2_[0m
[0m    12288 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads[0m
```

File: include/alpaka/core/Debug.hpp [hunk 2]
```diff
@@ -146,36 +146,3 @@ namespace alpaka
 #else
     #define ALPAKA_DEBUG_BREAK
 #endif
-
-//-----------------------------------------------------------------------------
```
Comment (psychocoderHPC) [inline include/alpaka/core/Debug.hpp]: This changes should not be a part of this pull request. Please move this changes to a new pull request.

**Comment by BenjaminW3** [comment]
Thanks for verifying this!
Both of your comments have been adressed.

**Comment by BenjaminW3** [comment]
push

**Comment by psychocoderHPC** [comment]
It looks like that it not influence the performace therefore it can be merged.

**Pull request merged**